src/sagemaker/xgboost/estimator.py

```diff
@@ -112,7 +111,7 @@ def __init__(
         )
 
         if py_version == "py2":
-            logger.warning(python_deprecation_warning(self.__framework_name__))
+            raise AttributeError("XGBoost container does not support Python 2, please use Python 3")
```

Comment:
what would happen before when someone tried XGBoost + Python 2?

Reply:
They will get errors when Python SDK tries to pull the xgboost py2 image because 1p algorithm team do not release xgboost py2 images.